### PR TITLE
XGBoost Evaluator Project Implementation

### PR DESCRIPTION
**Pull request by kindalime**
Hello! This is our cleaned-up implementation of the XGBoost Evaluator project, which was recently approved by the Google Team. Your input and feedback is much appreciated!

### CODE REVIEW

**Comment by cent5** [comment]
According to the proposal, this evaluator will be at `tfx_addons/xgboost_evaluator` and packaged. Has that plan changed?

https://github.com/tensorflow/tfx-addons/blob/main/proposals/20210707-xgboost_evaluator.md#project-directories

File: examples/sklearn_penguins/sklearn_predict_extractor_test.py
Comment (rcrowe-google) [inline examples/sklearn_penguins/sklearn_predict_extractor_test.py:142]: Please don't modify files in other projects without approval of the project owners.  Check the CODEOWNERS to see who owns each project.

**Review by rcrowe-google — COMMENTED** [review]
This project should be under https://github.com/tensorflow/tfx-addons/tree/main/tfx_addons/xgboost_evaluator

Including examples of usage is a good idea.  You can also place the examples under the project filder.

**Comment by rcrowe-google** [comment]
> According to the proposal, this evaluator will be at `tfx_addons/xgboost_evaluator` and packaged. Has that plan changed?
> 
> https://github.com/tensorflow/tfx-addons/blob/main/proposals/20210707-xgboost_evaluator.md#project-directories

That is correct.

**Comment by casassg** [comment]
@rcrowe-google we don't have write permissions for this repo, so we must use PRs unfortunately 😅

**Comment by casassg** [comment]
(also we should avoid pushing to main branch directly to avoid merge conflicts)

**Comment by casassg** [comment]
CC @rcrowe-google can you reopen the PR please 😄

File: examples/xgboost_evaluator/requirements.txt
Comment (casassg) [inline examples/xgboost_evaluator/requirements.txt:2]: Are we using scikit-learn?

**Review by casassg — COMMENTED** [review]
As mentioned by Robert, let's move this under tfx_addons/xgboost_evaluator

**Comment by rcrowe-google** [comment]
> @rcrowe-google we don't have write permissions for this repo, so we must use PRs unfortunately 😅

Right, forgot about that, sorry.  Also I don't think I fully grokked the power of codeowners, or lack thereof.

File: examples/xgboost_evaluator/requirements.txt
Comment (kindalime) [reply examples/xgboost_evaluator/requirements.txt]: Nope - that's been deleted.

Comment (kindalime) [reply examples/xgboost_evaluator/requirements.txt]: Nope - has been deleted.

File: setup.py
Comment (casassg) [inline setup.py:38]: nit: we should make this constraint wider to accept more versions of xgboost unless we use internal APIs.

File: tfx_addons/xgboost_evaluator/component.py
Comment (casassg) [inline tfx_addons/xgboost_evaluator/component.py:25]:
```suggestion
    super().__init__(module_file=xgboost_predict_extractor.get_module_file(), **kwargs)
```

File: tfx_addons/xgboost_evaluator/component.py
Comment (casassg) [inline tfx_addons/xgboost_evaluator/component.py:22]: Remember to add documentation to the component!

**Review by casassg — APPROVED** [review]

File: tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py
Comment (casassg) [inline tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py:131]: do we need to load this into a dataframe before passing it into DMatrix?

**Comment by rcrowe-google** [comment]
Let me know when you're ready for me to review and merge.

File: tfx_addons/xgboost_evaluator/component.py
Comment (casassg) [inline tfx_addons/xgboost_evaluator/component.py:24]:
```suggestion
  def __init__(self, **kwargs):
    if 'module_file' in kwargs:
      raise ValueError('XGBoostEvaluator does not accept custom module_file')
    super().__init__(module_file=xgboost_predict_extractor.get_module_file(), **kwargs)
```

File: tfx_addons/xgboost_evaluator/xgboost_predict_extractor_test.py
Comment (casassg) [inline tfx_addons/xgboost_evaluator/xgboost_predict_extractor_test.py:177]:
```suggestion
        example_splits=['eval'])
```

File: tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py
Comment (kindalime) [reply tfx_addons/xgboost_evaluator/xgboost_predict_extractor.py:131]: I don't believe so - the sklearn example does something similar to this in its example, and that works fine.

**Review by casassg — APPROVED** [review]
ready for wider review!

**Comment by kindalime** [comment]
@rcrowe-google Ready for review!

File: setup.py
Comment (hanneshapke) [inline setup.py:38]: #57 required 1.0.0. Should they all line up?
https://github.com/tensorflow/tfx-addons/pull/57/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R38

Comment (cent5) [reply setup.py:38]: If there is no error when setting this to `>=1.0.0`, I'd say let's use that. Major version 1 is supposed to have some level of backward compatibility.

Comment (kindalime) [reply setup.py]: Set to >=1.0.0.

**Review by rcrowe-google — APPROVED** [review]

**Pull request merged**
